sormas-ui/src/main/java/de/symeda/sormas/ui/utils/CheckBoxTreeUpdated.java

```diff
@@ -0,0 +1,240 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+import org.jetbrains.annotations.NotNull;
+
+import com.vaadin.server.Sizeable;
+import com.vaadin.shared.ui.MarginInfo;
+import com.vaadin.ui.Component;
+import com.vaadin.ui.HorizontalLayout;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.v7.data.util.converter.Converter;
+import com.vaadin.v7.ui.CheckBox;
+import com.vaadin.v7.ui.CustomField;
+
+public class CheckBoxTreeUpdated<ENUM extends Enum<?>> extends CustomField<Map<ENUM, Boolean>> {
+
+	private static final String[] INDENTATION_STYLES = new String[] {
+		CssStyles.INDENT_LEFT_1,
+		CssStyles.INDENT_LEFT_2,
+		CssStyles.INDENT_LEFT_3 };
+
+	private Map<ENUM, CheckBox> enumToggles = new HashMap<>();
+	private VerticalLayout checkBoxTreeLayout;
+	private HorizontalLayout checkBoxTreeLayoutForGroups;
+	private Class<ENUM> enumType;
+	private boolean addVerticalSpaces;
+	private UnaryOperator<ENUM> parentProvider;
+	private Function<ENUM, ? extends Enum> groupProvider;
+	private boolean settingToggles = false;
+	private int columns;
+
+	public CheckBoxTreeUpdated() {
+
+	}
+
+	public Map<ENUM, CheckBox> getEnumToggles() {
+		return enumToggles;
+	}
+
+	@Override
+	protected Component initContent() {
+
+		if (enumType != null) {
+			if (groupProvider == null) {
+				checkBoxTreeLayout = new VerticalLayout();
+				checkBoxTreeLayout.setMargin(false);
+				checkBoxTreeLayout.setSpacing(false);
+				checkBoxTreeLayout.setWidth(100, Sizeable.Unit.PERCENTAGE);
+				addCheckBoxes();
+				final MarginInfo marginInfo = new MarginInfo(false, false, true, false);
+				checkBoxTreeLayout.setMargin(marginInfo);
+			} else {
+				checkBoxTreeLayoutForGroups = new HorizontalLayout();
+				checkBoxTreeLayoutForGroups.setMargin(false);
+				checkBoxTreeLayoutForGroups.setWidth(100, Unit.PERCENTAGE);
+				addGroups();
+			}
+
+			settingToggles = true;
+			setToggleValues(getValue());
+			settingToggles = false;
+		}
+
+		addValueChangeListener(valueChangeEvent -> {
+			Map<ENUM, Boolean> value = (Map<ENUM, Boolean>) valueChangeEvent.getProperty().getValue();
+			final Map<ENUM, Boolean> nullSafeValue = value == null ? new HashMap<>() : value;
+			settingToggles = true;
+			enumToggles.forEach(((anEnum, checkBox) -> {
+				final Boolean orDefault = nullSafeValue.getOrDefault(anEnum, false);
+				if (!Objects.equals(checkBox.getValue(), orDefault)) {
+					checkBox.setValue(orDefault);
+				}
+			}));
+			settingToggles = false;
+		});
+
+		if (checkBoxTreeLayout != null) {
+			return checkBoxTreeLayout;
+		}
+
+		return checkBoxTreeLayoutForGroups;
+	}
+
+	private void addGroups() {
+		final ENUM[] enumElements = enumType.getEnumConstants();
+
+		Map<Enum, List<ENUM>> enumGroupMap = new HashMap<>();
```

Comment:
please use the Stream api to create the groups

e.g. `.collect(groupingBy(....)`

Reply:
changed approach

Reply:
not resolved

Reply:
updated